api/provider/aws/load_balancer_delete.go

```diff
@@ -0,0 +1,91 @@
+package aws
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/elb"
+	"github.com/aws/aws-sdk-go/service/iam"
+)
+
+func (l *LoadBalancerProvider) Delete(loadBalancerID string) error {
+	fqLoadBalancerID := addLayer0Prefix(l.Config.Instance(), loadBalancerID)
+
+	if err := l.deleteLoadBalancer(fqLoadBalancerID); err != nil {
+		return err
+	}
+
+	roleName := fmt.Sprintf("%s-lb", fqLoadBalancerID)
```

Comment:
nit: I think when constructing a role (or sg), even though it is simple, would be good to separate into a method.